mevislab.github.io/content/tutorials/summary/summary3.md

```diff
@@ -0,0 +1,691 @@
+
+{{< highlight filename="<MACRO_NAME>.script" >}}
+```Stan
+Commands {
+  source = $(LOCAL)/TutorialSummary.py
+  
+  FieldListener selected3DView {
+    command = viewSelectionChanged
+  }
+}
+```
+{{</highlight>}}
+
+Right-click {{< mousebutton "right" >}} the command and select {{< menuitem "Create Python Function 'viewSelectionChanged'" >}}. MATE automatically opens the Python file of your Macro Module and creates a function *viewSelectionChanged*.
+
+{{< highlight filename="<MACRO_NAME>.py" >}}
+```Python
+from mevis import *
+
+def viewSelectionChanged(field):
+  if field.value == "Segmentation":
+    ctx.field("SoSwitch.whichChild").value = 0
+  if field.value == "Image":
+    ctx.field("SoSwitch.whichChild").value = 1
+  if field.value == "Both":
+    ctx.field("SoSwitch.whichChild").value = 2
+```
+{{</highlight>}}
+
+The function sets the `SoSwitch` to the child value depending on the selected *Field* value from the *ComboBox* and you should now be able to switch the 3D rendering by selecting an entry in the User Interface.
+
+#### Setting the Marker
+The Marker for the `RegionGrowing` is defined by the click position as Vector3. Add another *Field markerPosition* to the *Parameters* section and define *type = Vector3*.
+
+Teen, add a trigger *Field* *applyMarker* to your *Parameters* section. Set *type  = Trigger* and *title = Add*.
+
+{{< highlight filename="<MACRO_NAME>.script" >}}
+```Stan
+...
+    Field markerPosition {
+      type         = Vector3
+    }
+    Field applyMarker {
+      type         = Trigger
+      title        = Add
+    }
+...
+```
+{{</highlight>}}
+
+Add another *FieldListener* to both fields:
+{{< highlight filename="<MACRO_NAME>.script" >}}
+```Stan
+...
+  FieldListener markerPosition {
+    command = insertPosition
+  }
+  FieldListener applyMarker {
+    command = applyPosition
+  }
+...
+```
+{{</highlight>}}
+
+In the end, add both fields to the *Settings* section of your user interface:
+{{< highlight filename="<MACRO_NAME>.script" >}}
+```Stan
+...
+      Field markerPosition {}
+      Field applyMarker {}
+...
+```
+{{</highlight>}}
+
+The Python functions should look like this:
+{{< highlight filename="<MACRO_NAME>.py" >}}
+```Python
+...
+def insertPosition(field):
+  ctx.field("SoView2DMarkerEditor.newPosXYZ").value = field.value
+
+def applyPosition():
+  ctx.field("SoView2DMarkerEditor.useInsertTemplate").value = True
+  ctx.field("SoView2DMarkerEditor.add").touch()
+...
+```
+{{</highlight>}}
+
+Whenever the Field *markerPosition* changes its value, the value is automatically applied to the *SoView2DMarkerEditor.newPosXYZ*. Clicking *SoView2DMarkerEditor.add* adds the new Vector to the `SoView2DMarkerEditor` and the region growing starts.
+
+{{<alert class="info" caption="Info">}}
+The *Field* *SoView2DMarkerEditor.useInsertTemplate* needs to be set to *True* in order to allow adding Markers via Python.
```

Comment:
Markers -> markers
I only occasionally commented on this, but there are many occurrences of capitalized words that should not be capitalized, because they are not names. Nitpicking...